src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/page.tsx

```diff
@@ -88,6 +88,9 @@ export default async function SettingsPage({ searchParams }: Props) {
   );
 
   const userData = await getSubscriberById(session.user.subscriber.id);
+  if (!userData) {
```

Comment:
A couple of the subscriber-fetching functions (e.g. `getSubscriberById`) assumed that they'd always find one in the DB, but theoretically you could have an invalid ID in here. I thought it was safest to keep that exposed in the types and handle it at the call sites - such as here.

So _theoretically_ this should never get hit, but if someone ends up with a session with an invalid subscriber ID and tries to visit the settings page, this will redirect them to the landing page.